### PR TITLE
sign up test

### PR DESCRIPTION
**Pull request by ricardoamoreno**
## Summary

sign up test finished

- test: succesfully case
- test: email already used
- test: invalid password
- test: firstName missing
- test: lastName missing
- test: email missing
- test: password missing

## Known Issues

none

## Trello Card
[Test de Sign Up](https://trello.com/c/oIpSsLgw)

### CODE REVIEW

File: test/setup.js [hunk 1]
```diff
@@ -7,6 +7,6 @@ const truncateTable = model =>
 
 const truncateDatabase = () => Promise.all(tables.map(truncateTable));
 
-global.beforeEach(async () => {
+global.beforeAll(async () => {
```
Comment (martinremis) [inline test/setup.js]: Why is this change?

File: test/signup.spec.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  it('should respond succesfully if it passes all validations', () =>
+    createUser(mockUser).then(res => {
+      expect(res.statusCode).toBe(201);
+      expect(res.body).toHaveProperty('userId');
+    }));
+
+  it('should fail with error if email already exist', () =>
+    createUser(mockUser).then(res => {
+      expect(res.statusCode).toBe(503);
+      expect(res.body.internal_code).toContain('database_error');
```
Comment (martinremis) [inline test/signup.spec.js]: You can use the const for `database_error` (stored in errors middleware if I remember right)

File: test/signup.spec.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }));
+
+  it('should fail with validation error if pass invalid password', () =>
+    createUser(mockUserInvalidPass).then(res => {
+      expect(res.statusCode).toBe(400);
+      expect(res.body.internal_code).toContain('validation_error');
```
Comment (martinremis) [inline test/signup.spec.js]: Same

File: test/signup.spec.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }));
+
+  it('should fail whit validation error if firsName is missing', () =>
+    createUser(mockUserMissingFirstName).then(res => {
+      expect(res.statusCode).toBe(400);
+      expect(res.body.internal_code).toContain('validation_error');
```
Comment (martinremis) [inline test/signup.spec.js]: Same

File: test/signup.spec.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      expect(res.body.message[0].message).toContain("must have required property 'firstName'");
```
Comment (martinremis) [inline test/signup.spec.js]: Is this string defined by a library? Do we have a constant to set this message?

File: test/signup.spec.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    }));
+
+  it('should fail whit validation error if lastName is missing', () =>
+    createUser(mockUserMissingLastName).then(res => {
+      expect(res.statusCode).toBe(400);
+      expect(res.body.internal_code).toContain('validation_error');
```
Comment (martinremis) [inline test/signup.spec.js]: Same

File: test/signup.spec.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+      expect(res.body.message[0].message).toContain("must have required property 'lastName'");
```
Comment (martinremis) [inline test/signup.spec.js]: Same

File: test/signup.spec.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    }));
+  it('should fail whit validation error if email is missing', () =>
+    createUser(mockUserMissingEmail).then(res => {
+      expect(res.statusCode).toBe(400);
+      expect(res.body.internal_code).toContain('validation_error');
```
Comment (martinremis) [inline test/signup.spec.js]: Same

File: test/signup.spec.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+      expect(res.body.message[0].message).toContain("must have required property 'email'");
+    }));
+
+  it('should fail whit validation error if password is missing', () =>
+    createUser(mockUserMissingPassword).then(res => {
+      expect(res.statusCode).toBe(400);
+      expect(res.body.internal_code).toContain('validation_error');
```
Comment (martinremis) [inline test/signup.spec.js]: Same

File: test/signup.spec.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    createUser(mockUserMissingLastName).then(res => {
+      expect(res.statusCode).toBe(400);
+      expect(res.body.internal_code).toContain('validation_error');
+      expect(res.body.message[0].message).toContain("must have required property 'lastName'");
+    }));
+  it('should fail whit validation error if email is missing', () =>
+    createUser(mockUserMissingEmail).then(res => {
+      expect(res.statusCode).toBe(400);
+      expect(res.body.internal_code).toContain('validation_error');
+      expect(res.body.message[0].message).toContain("must have required property 'email'");
```
Comment (martinremis) [inline test/signup.spec.js]: Same

File: test/signup.spec.js [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    }));
+
+  it('should fail whit validation error if password is missing', () =>
+    createUser(mockUserMissingPassword).then(res => {
+      expect(res.statusCode).toBe(400);
+      expect(res.body.internal_code).toContain('validation_error');
+      expect(res.body.message[0].message).toContain("must have required property 'password'");
```
Comment (martinremis) [inline test/signup.spec.js]: Same

**Review by martinremis — CHANGES_REQUESTED** [review]
All the schema tests (missing property tests) could have reused code. It might not be so relevant here, but it will be if the request body has 15 properties xD

File: test/setup.js (hunk 1 above)
Comment (ricardoamoreno) [reply test/setup.js]: I change it to persist the user data and then test that the email already exists in the database

File: test/signup.spec.js (hunk 5 above)
Comment (ricardoamoreno) [reply test/signup.spec.js]: Yes, it's the answer that defines ajv, I arranged it that way to differentiate each parameter that does not pass the validation, I don't know, is it wrong?

File: test/setup.js (hunk 1 above)
Comment (ricardoamoreno) [reply test/setup.js]: I restore it and modify the specific test

File: test/signup.spec.js (hunk 2 above)
Comment (ricardoamoreno) [reply test/signup.spec.js]: change it to the error constant

File: test/signup.spec.js (hunk 3 above)
Comment (ricardoamoreno) [reply test/signup.spec.js]: i changed it to error constant

File: test/signup.spec.js (hunk 4 above)
Comment (ricardoamoreno) [reply test/signup.spec.js]: i changed it to error constant

File: test/signup.spec.js (hunk 7 above)
Comment (ricardoamoreno) [reply test/signup.spec.js]: i changed it to verify missingProperty

File: test/signup.spec.js (hunk 9 above)
Comment (ricardoamoreno) [reply test/signup.spec.js]: i changed it to verify missingProperty

File: test/signup.spec.js (hunk 11 above)
Comment (ricardoamoreno) [reply test/signup.spec.js]: i changed it to verify missingProperty

File: test/signup.spec.js (hunk 8 above)
Comment (ricardoamoreno) [reply test/signup.spec.js]: i changed it to verify missingProperty

File: test/signup.spec.js (hunk 6 above)
Comment (ricardoamoreno) [reply test/signup.spec.js]: i changed it to verify missingProperty

File: test/mocks/mockUsers.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
+exports.mockUser = {
+  firstName: 'ricardo',
+  lastName: 'moreno',
+  email: 'ricardo.a.moreno@wolox.co',
```
Comment (martinremis) [inline test/mocks/mockUsers.js]: Remove any real people information from a public repository (or even a private one)

**Review by martinremis — CHANGES_REQUESTED** [review]

File: test/mocks/mockUsers.js (hunk 12 above)
Comment (ricardoamoreno) [reply test/mocks/mockUsers.js]: I replaced it with generic user information

**Review by martinremis — APPROVED** [review]

**Pull request merged**